### ISSUE TITLE
eksctl 0.86 does not respect availability zone specification

### ISSUE THREAD

**Issue by NukaCody** [weaveworks/eksctl#4906]
**What were you trying to accomplish?**

Deploy an EKS cluster to specific availability zones

**What happened?**

eksctl chooses random AZs

**How to reproduce it?**

I ran `eksctl create cluster --verbose 5 -f mythicaleks.yaml` where the yaml config is:

```yaml
---
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: mythicaleks-eksctl
  region: us-east-1
  version: "1.19"

availabilityZones: ["us-east-1a", "us-east-1b", "us-east-1c"]

managedNodeGroups:
- name: nodegroup
  desiredCapacity: 3
  ssh:
    allow: true
    publicKeyName: mythicaleks

# To enable all of the control plane logs, uncomment below:
# cloudWatch:
#  clusterLogging:
#    enableTypes: ["*"]
```

I did this with both 0.85 and 0.86. Below is the log for 0.86

```text
2022-03-04 19:33:51 [▶]  role ARN for the current session is "arn:aws:sts::453244917115:assumed-role/TeamRole/MasterKey"
2022-03-04 19:33:51 [ℹ]  eksctl version 0.86.0
2022-03-04 19:33:51 [ℹ]  using region us-east-1
2022-03-04 19:33:51 [▶]  determining availability zones
2022-03-04 19:33:51 [ℹ]  setting availability zones to [us-east-1f us-east-1a]
2022-03-04 19:33:51 [▶]  VPC CIDR (192.168.0.0/16) was divided into 8 subnets [192.168.0.0/19 192.168.32.0/19 192.168.64.0/19 192.168.96.0/19 192.168.128.0/19 192.168.160.0/19 192.168.192.0/19 192.168.224.0/19]
2022-03-04 19:33:51 [ℹ]  subnets for us-east-1f - public:192.168.0.0/19 private:192.168.64.0/19
2022-03-04 19:33:51 [ℹ]  subnets for us-east-1a - public:192.168.32.0/19 private:192.168.96.0/19
2022-03-04 19:33:51 [ℹ]  nodegroup "nodegroup" will use "" [AmazonLinux2/1.19]
2022-03-04 19:33:51 [ℹ]  using EC2 key pair "mythicaleks"
2022-03-04 19:33:51 [ℹ]  using Kubernetes version 1.19
2022-03-04 19:33:51 [ℹ]  creating EKS cluster "mythicaleks-eksctl" in "us-east-1" region with managed nodes
```

Here is the log for 0.85

```text
TeamRole:~/environment/amazon-ecs-mythicalmysfits-workshop/workshop-1 (master) $ eksctl create cluster --verbose 5 -f mythicaleks.yaml
2022-03-04 19:39:55 [▶]  role ARN for the current session is "arn:aws:sts::453244917115:assumed-role/TeamRole/MasterKey"
2022-03-04 19:39:55 [ℹ]  eksctl version 0.85.0
2022-03-04 19:39:55 [ℹ]  using region us-east-1
2022-03-04 19:39:55 [▶]  VPC CIDR (192.168.0.0/16) was divided into 8 subnets [192.168.0.0/19 192.168.32.0/19 192.168.64.0/19 192.168.96.0/19 192.168.128.0/19 192.168.160.0/19 192.168.192.0/19 192.168.224.0/19]
2022-03-04 19:39:55 [ℹ]  subnets for us-east-1a - public:192.168.0.0/19 private:192.168.96.0/19
2022-03-04 19:39:55 [ℹ]  subnets for us-east-1b - public:192.168.32.0/19 private:192.168.128.0/19
2022-03-04 19:39:55 [ℹ]  subnets for us-east-1c - public:192.168.64.0/19 private:192.168.160.0/19
2022-03-04 19:39:55 [ℹ]  nodegroup "nodegroup" will use "" [AmazonLinux2/1.19]
2022-03-04 19:39:55 [ℹ]  using EC2 key pair "mythicaleks"
2022-03-04 19:39:55 [ℹ]  using Kubernetes version 1.19
2022-03-04 19:39:55 [ℹ]  creating EKS cluster "mythicaleks-eksctl" in "us-east-1" region with managed nodes
```

**Comment by aclevername** [comment]
I believe this was fixed in 0.87.0-rc0, can you give that a try please 😄 ?

**Comment by NukaCody** [comment]
Sure can, looks like 0.87.0-rc.0 fixes it:

```text
2022-03-04 20:08:36 [▶]  role ARN for the current session is "arn:aws:sts::453244917115:assumed-role/TeamRole/MasterKey"
2022-03-04 20:08:36 [ℹ]  eksctl version 0.87.0-rc.0
2022-03-04 20:08:36 [ℹ]  using region us-east-1
2022-03-04 20:08:36 [▶]  VPC CIDR (192.168.0.0/16) was divided into 8 subnets [192.168.0.0/19 192.168.32.0/19 192.168.64.0/19 192.168.96.0/19 192.168.128.0/19 192.168.160.0/19 192.168.192.0/19 192.168.224.0/19]
2022-03-04 20:08:36 [ℹ]  subnets for us-east-1a - public:192.168.0.0/19 private:192.168.96.0/19
2022-03-04 20:08:36 [ℹ]  subnets for us-east-1b - public:192.168.32.0/19 private:192.168.128.0/19
2022-03-04 20:08:36 [ℹ]  subnets for us-east-1c - public:192.168.64.0/19 private:192.168.160.0/19
2022-03-04 20:08:36 [ℹ]  nodegroup "nodegroup" will use "" [AmazonLinux2/1.19]
2022-03-04 20:08:36 [ℹ]  using EC2 key pair "mythicaleks"
2022-03-04 20:08:36 [ℹ]  using Kubernetes version 1.19
2022-03-04 20:08:36 [ℹ]  creating EKS cluster "mythicaleks-eksctl" in "us-east-1" region with managed nodes
```
